src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts

```diff
@@ -47,24 +45,40 @@ export const createEnsureDefaultIndexPattern = (
     if (canUpdateUiSetting === false) {
       return;
     }
-    const patterns = await this.getIds();
-    let defaultId = await uiSettings.get('defaultIndex');
-    let defined = !!defaultId;
-    const exists = includes(patterns, defaultId);
```

Comment:
These logic should not be removed I think, it is checking whether the defaultIndex is in the list of index patterns.